query/src/main/java/org/infinispan/query/dsl/embedded/impl/HibernateSearchPropertyHelper.java

```diff
@@ -149,6 +142,15 @@ private Object convertToPropertyType(String value, FieldBridge bridge) {
       }
    }
 
+   private boolean isElasticsearchDateBridge(FieldBridge bridge) {
+      try {
+         return bridge instanceof org.hibernate.search.elasticsearch.bridge.builtin.impl.ElasticsearchDateBridge;
+      } catch (Exception e) {
+         // this happens because Elastic search could be missing from classpath, being an optional dependency
+         return false;
```

Comment:
I'm not sure why this is preferable to having the check in the constructor? With the constructor approach, we only have to catch a Exception once. Whereas now an exception is potentially thrown per invocation of `convertToPropertyType`

Reply:
I'm fine with any approach that does not require the ClassLoader parameter to those constructors. The change was a bit disruptive. So to mitigate the 'per invocation' we can add the flag back.